### PR TITLE
Support toc generation in the api-documenter

### PR DESCRIPTION
**Pull request by Feiyang1**
To generate a toc, use command `api-documenter-fire toc -i temp -p "/docs/reference/js/v9"`. `temp` is the folder that contains all api.json files.

The command will create a file `toc/toc.yaml`. To change the output folder, use the flag `-o`.

### CODE REVIEW

**Review by hsubox76 — APPROVED** [review]

File: repo-scripts/api-documenter/README.md [hunk 1]
```diff
@@ -3,3 +3,14 @@
 It is a fork of [API Documenter](https://github.com/microsoft/rushstack/tree/master/apps/api-documenter)
 It reads the *.api.json data files produced by [API Extractor](https://api-extractor.com/),
 and then generates files in [Markdown](https://en.wikipedia.org/wiki/Markdown) format suitable for displaying in Firebase Devsite.
+
+## Generate toc for Firebase devsite
```
Comment (hiranya911) [inline repo-scripts/api-documenter/README.md:7]: You might want to document  the `-g3-path` flag here as well.

File: repo-scripts/api-documenter/src/cli/TocAction.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import { BaseAction } from './BaseAction';
+import { generateToc } from '../toc';
+
+export class TocAction extends BaseAction {
+  private _g3PathParameter!: CommandLineStringParameter;
+  private _jsSDKParameter!: CommandLineFlagParameter;
+  public constructor(parser: ApiDocumenterCommandLine) {
+    super({
+      actionName: 'toc',
+      summary: 'Generate documentation as Markdown files (*.md)',
```
Comment (hiranya911) [inline repo-scripts/api-documenter/src/cli/TocAction.ts:32]: Update text to reflect the purpose of this action.

File: repo-scripts/api-documenter/src/cli/TocAction.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        'Generates API documentation as a collection of files in' +
+        ' Markdown format, suitable for example for publishing on a GitHub site.'
+    });
+  }
+
+  protected onDefineParameters(): void {
+    super.onDefineParameters();
+
+    this._g3PathParameter = this.defineStringParameter({
+      parameterLongName: '--g3-path',
```
Comment (hiranya911) [inline repo-scripts/api-documenter/src/cli/TocAction.ts:43]: How about calling this `--host-path` or `--site-dir` (something that's not g3 specific)?

**Review by hiranya911 — COMMENTED** [review]
Looks pretty good. Just a couple of suggestions.

File: repo-scripts/api-documenter/README.md (hunk 1 above)
Comment (Feiyang1) [reply repo-scripts/api-documenter/README.md:7]: -p is the short flag for --g3-path, but good point I will list both of them

File: repo-scripts/api-documenter/src/cli/TocAction.ts (hunk 3 above)
Comment (Feiyang1) [reply repo-scripts/api-documenter/src/cli/TocAction.ts:43]: changed to `--host-path`

**Pull request merged**